pkg/app/utils.go

```diff
@@ -47,27 +50,29 @@ func EnsureAppProjectName(userNSClient v1.NamespaceInterface, ownedProjectID, cl
 		}
 	}
 
-	// move Namespace into a project
 	expectedAppProjectName := fmt.Sprintf("%s:%s", clusterName, ownedProjectID)
-	appProjectName := ""
-	if projectName, ok := deployNamespace.Annotations[nslabels.ProjectIDFieldLabel]; ok {
-		appProjectName = projectName
+	appProjectID := deployNamespace.Annotations[nslabels.ProjectIDFieldLabel]
+	if appProjectID != "" && slice.ContainsString(strings.Split(settings.SystemNamespaces.Get(), ","), deployNamespace.Name, nil) {
```

Comment:
Should we be using the setting or the annotation since some are added dynamically like fleet

Reply:
Is there an annotation that would say that this namespace is a system namespace?

Reply:
Yes, `management.cattle.io/system-namespace`

Reply:
@StrongMonkey How are we marking fleet namespaces as system namespaces

Reply:
We check if label `fleet.cattle.io/managed` is true for namespaces. All the generated namespaces from fleet have this label.
https://github.com/rancher/rancher/blob/4b967ddcf26f87aa5ce1bdf5932d9a1dd5cbbe92/pkg/controllers/managementuser/rbac/namespace_handler.go#L172

Reply:
Updated.